plugins/generator-1.20.1/forge-1.20.1/templates/overlay.java.ftl

```diff
@@ -79,34 +76,30 @@ package ${package}.client.screens;
             </#if>
 
             <#list data.getComponentsOfType("Image") as component>
-                <#assign x = component.x - 213>
-                <#assign y = component.y - 120>
                 <#if hasProcedure(component.displayCondition)>
                         if (<@procedureOBJToConditionCode component.displayCondition/>) {
                 </#if>
-                    event.getGuiGraphics().blit(new ResourceLocation("${modid}:textures/screens/${component.image}"), posX + ${x}, posY + ${y}, 0, 0,
+                    event.getGuiGraphics().blit(new ResourceLocation("${modid}:textures/screens/${component.image}"), <@calculatePosition component/>, 0, 0,
                         ${component.getWidth(w.getWorkspace())}, ${component.getHeight(w.getWorkspace())},
                         ${component.getWidth(w.getWorkspace())}, ${component.getHeight(w.getWorkspace())});
                 <#if hasProcedure(component.displayCondition)>}</#if>
             </#list>
 
             <#list data.getComponentsOfType("Label") as component>
-                <#assign x = component.x - 213>
-                <#assign y = component.y - 120>
-                    <#if hasProcedure(component.displayCondition)>
-                        if (<@procedureOBJToConditionCode component.displayCondition/>)
-                    </#if>
-                    event.getGuiGraphics().drawString(Minecraft.getInstance().font,
-                        <#if hasProcedure(component.text)><@procedureOBJToStringCode component.text/><#else>Component.translatable("gui.${modid}.${registryname}.${component.getName()}")</#if>,
-                        posX + ${x}, posY + ${y}, ${component.color.getRGB()}, false);
+                <#if hasProcedure(component.displayCondition)>
+                    if (<@procedureOBJToConditionCode component.displayCondition/>)
+                </#if>
+                event.getGuiGraphics().drawString(Minecraft.getInstance().font,
+                    <#if hasProcedure(component.text)><@procedureOBJToStringCode component.text/><#else>Component.translatable("gui.${modid}.${registryname}.${component.getName()}")</#if>,
+                    <@calculatePosition component/>, ${component.color.getRGB()}, false);
             </#list>
 
 			<#list data.getComponentsOfType("EntityModel") as component>
 			    if (<@procedureOBJToConditionCode component.entityModel/> instanceof LivingEntity livingEntity) {
 			    	<#if hasProcedure(component.displayCondition)>
                         if (<@procedureOBJToConditionCode component.displayCondition/>)
                     </#if>
-			        InventoryScreen.renderEntityInInventoryFollowsAngle(event.getGuiGraphics(), posX + ${component.x - 202}, posY + ${component.y - 100},
+					InventoryScreen.renderEntityInInventoryFollowsAngle(event.getGuiGraphics(), <@calculatePosition component/>,

```

Comment:
Those now no longer have the right offset of 202 and 100, you will need to introduce optional parameters to the macro to take those offsets into account